src/Microsoft.DotNet.Build.CloudTestTasks/PackageFiles/CloudTest.targets

```diff
@@ -96,25 +97,66 @@
+      <OSPlatformTestArchives Include="$(TestArchivesRoot)**/*.zip" >
         <OSPlatformConfig>$(OSPlatformConfig)</OSPlatformConfig>
-      </UnfilteredTestArchives>
-      <UnfilteredTestArchives Include="$(AnyOSTestArchivesRoot)**/*.zip" >
+      </OSPlatformTestArchives>
+      <AnyOSPlatformTestArchives Include="$(AnyOSTestArchivesRoot)**/*.zip" >
         <OSPlatformConfig>$(AnyOSPlatformConfig)</OSPlatformConfig>
-      </UnfilteredTestArchives>
+      </AnyOSPlatformTestArchives>
       <!-- Only include Unix folders if supported by the current target OS -->
-      <UnfilteredTestArchives Condition="'$(TargetsUnix)' == 'true'" Include="$(UnixTestArchivesRoot)**/*.zip" >
+      <UnixPlatformTestArchives Condition="'$(TargetsUnix)' == 'true'" Include="$(UnixTestArchivesRoot)**/*.zip" >
         <OSPlatformConfig>$(UnixPlatformConfig)</OSPlatformConfig>
-      </UnfilteredTestArchives>
+      </UnixPlatformTestArchives>
+    </ItemGroup>
+
+    <PropertyGroup>
+        <OSPlatformTestArchivesIndexFile>$(TestArchivesRoot)/$(OSPlatformConfig)-Test-Archive-Index.txt</OSPlatformTestArchivesIndexFile>
+        <AnyOSPlatformTestArchivesIndexFile>$(AnyOSTestArchivesRoot)/$(AnyOSPlatformConfig)-Test-Archive-Index.txt</AnyOSPlatformTestArchivesIndexFile>
+        <UnixPlatformTestArchivesIndexFile>$(UnixTestArchivesRoot)/$(UnixPlatformConfig)-Test-Archive-Index.txt</UnixPlatformTestArchivesIndexFile>
+    </PropertyGroup>
+
+    <!-- This is where we craft our archive indexes. These will sit at the base of the uploaded package. -->
+    <WriteLinesToFile 
+        File="$(OSPlatformTestArchivesIndexFile)"
+        Lines="@(OSPlatformTestArchives -> '%(Filename)%(Extension)')"
+        Overwrite="true"
+        Encoding="Unicode" />
+    
+    <WriteLinesToFile 
+        File="$(AnyOSPlatformTestArchivesIndexFile)"
+        Lines="@(UnixPlatformTestArchives -> '%(Filename)%(Extension)')"
+        Overwrite="true"
+        Encoding="Unicode" />
+
+    <WriteLinesToFile 
+        File="$(UnixPlatformTestArchivesIndexFile)"
+        Lines="@(UnixPlatformTestArchives -> '%(Filename)%(Extension)')"
+        Overwrite="true"
+        Encoding="Unicode" />
+   
+    <ItemGroup>
+      <ForUpload Include="$(OSPlatformTestArchivesIndexFile)" Condition="Exists('$(OSPlatformTestArchivesIndexFile)')"/>
+      <ForUpload Include="$(AnyOSPlatformTestArchivesIndexFile)" Condition="Exists('$(AnyOSPlatformTestArchivesIndexFile)')"/>
+      <ForUpload Include="$(UnixPlatformTestArchivesIndexFile)" Condition="Exists('$(UnixPlatformTestArchivesIndexFile)')" />
+    </ItemGroup>
+
+   <!-- The unfiltered tests is the union of all of the tests archives we've selected -->
+    <ItemGroup>   
+      <UnfilteredTestArchives Include="@(OSPlatformTestArchives);@(UnixPlatformTestArchives);@(UnixPlatformTestArchives)" />
     </ItemGroup>
+
+    <Message Text="Full, Unfiltered Test Archive Collection :: @(UnfilteredTestArchives)" Importance="Low" />
```

Comment:
Any way to make it diagnostic? Otherwise I can just remove it. (It is incredibly verbose).